### DISCUSSION TITLE
PropertyChanged not thrown

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#15209] [Q&A] [answered]
Hello,

I have a window which is separated into two sections, a right and a left one. In the left there is a table (DataGrid) of items. If you select an item, details about the object behind the SelectedItem should be shown on the right in TextBox(es). This is working as intended.

The property SelectedItem.Name on the left gives the text for the TextBox.Content on the right.

Changes on the right side of the property "Content" should be reflected instantaneously also on the left in the DataGrid for the SelectedItem. This is not working.

I tried it this way:

```  
    private ProjectsTaskEntity? _selecteditem;

    // binded to the SelectedItem property on the left (DataGrid)
    public ProjectsTaskEntity? SelectedItem
    {
        get => _selecteditem;
        set
        {
            this.RaiseAndSetIfChanged(ref _selecteditem, value); 
            this.RaisePropertyChanged("SelectedName");
        }
    }

    // binded to TextBox.Content on the right
    public string? SelectedName
    {
        get => SelectedItem?.Name;
        set
        {
            if (SelectedItem is not null)
            {
                SelectedItem!.Name = value;
                this.RaisePropertyChanged();
                this.RaisePropertyChanged("SelectedItem");
            }
        }
    }
```

If I alter the text on the right in the TextBox, the setter of property SelectedName is called and the view is updated but only the Textbox, not the item in the DataGrid. Obviously ```this.RaisePropertyChanged("SelectedItem")```  has no effect.

By altering the TextBox content, the setter of SeletedItem is called and _selecteditem is altered. But the getter is not called, although ```RaisePropertyChanged("SelectedItem")``` is triggered. It seems that the content of the SelectedItem is not updated by the event but only the state and the highlighting of the corresponding row.

If this assumption is correct, how to set up such a cross linking between two controls to get the functionality as described above?

**Comment by timunie** [comment]
SelectedItem.Name needs to raise proeprty changes for Name

**Reply by drei97** [reply]
do you mean like this:

```
this.RaisePropertyChanged("SelectedItem.Name");
```

I already tried it with no success.

**Reply by timunie** [reply, marked as answer]
like that:

```cs
public class SelectedItem : ReactiveObject
{
    private string _Name; 
    public string Name 
    { 
       get => _Name;
       set => this.RaiseAndSetIfChanged(ref _Name, value);
    }
}
```

**Reply by drei97** [reply]
I got you.

the class of the SelectedItem object must be inherited from ReactiveObject and the property must raise the event from inside that object.

thnx
